tests/next_tests/unit_tests/iterator_tests/transforms_tests/test_domain_inference.py

```diff
@@ -0,0 +1,172 @@
+                                 size_field=[im.literal_from_value(11), im.literal_from_value(7), im.literal_from_value(3)])
+
+    expected = im.call(im.call('as_fieldop')(stencil, domain))(im.ref('in_field1'), im.ref('in_field2'), im.ref('in_field3'))
+    expected_domains = {'in_field1': SymbolicDomain.from_expr(
+        im.cartesian_domain({"IDim": (1, 1), "JDim": (0, 0), "KDim": (0, 0)}, "tmp",
+                            size_field=[im.literal_from_value(11), im.literal_from_value(7), im.literal_from_value(3)])),
+                        'in_field2': SymbolicDomain.from_expr(
+        im.cartesian_domain({"IDim": (0, 0), "JDim": (1, 1), "KDim": (0, 0)}, "tmp",
+                            size_field=[im.literal_from_value(11), im.literal_from_value(7), im.literal_from_value(3)])),
+                        'in_field3': SymbolicDomain.from_expr(
+        im.cartesian_domain({"IDim": (0, 0), "JDim": (0, 0), "KDim": (-1, -1)}, "tmp",
+                            size_field=[im.literal_from_value(11), im.literal_from_value(7), im.literal_from_value(3)]))
+                        }
+
+    actual_call, actual_domains = InferDomain.infer_as_fieldop(testee, domain)
+
+    assert actual_call == expected
+    assert actual_domains == expected_domains
+
+
+def test_nested_stencils():
+    inner_stencil = im.lambda_('arg0_tmp', 'arg1_tmp')(im.plus( im.deref(im.shift(itir.SymbolRef('Ioff'), 1)(
+                     "arg0_tmp")),im.deref(im.shift(itir.SymbolRef('Joff'), -1)("arg1_tmp"))))
+    stencil = im.lambda_('arg0', 'arg1')(im.plus( im.deref(im.shift(itir.SymbolRef('Ioff'), 1)(
+                     "arg0")),im.deref(im.shift(itir.SymbolRef('Joff'), -1)("arg1"))))
+
+    tmp = im.call(im.call('as_fieldop')(inner_stencil))(im.ref('in_field1'), im.ref('in_field2'))
+
+    testee = im.call(im.call('as_fieldop')(stencil))(im.ref('in_field1'), tmp)
+
+    domain_inner = im.cartesian_domain({"IDim": (0, 0), "JDim": (-1, -1)}, "tmp", size_field=[im.literal_from_value(11),im.literal_from_value(7)])
+    domain = im.cartesian_domain({"IDim": (0, 0), "JDim": (0, 0)}, "tmp", size_field=[im.literal_from_value(11),im.literal_from_value(7)])
+
+    expected = im.call(im.call('as_fieldop')( stencil, domain))(im.ref('in_field1'),im.call(im.call('as_fieldop')(inner_stencil, domain_inner))(im.ref('in_field1'), im.ref('in_field2')))
+    expected_domains = {'in_field1': SymbolicDomain.from_expr(im.cartesian_domain({"IDim": (1, 1), "JDim": (-1, 0)}, "tmp", size_field=[im.literal_from_value(11),im.literal_from_value(7)])),
+                        'in_field2': SymbolicDomain.from_expr(im.cartesian_domain({"IDim": (0, 0), "JDim": (-2, -2)}, "tmp", size_field=[im.literal_from_value(11),im.literal_from_value(7)]))}
+
+    actual_call, actual_domains = InferDomain.infer_as_fieldop(testee, domain)
+
+    assert actual_call == expected
+    assert actual_domains == expected_domains
+
+def test_nested_stencils_n_times():
```

Comment:
You can parameterize this test (https://docs.pytest.org/en/7.0.x/how-to/parametrize.html) and which should make it a little simpler because you don't need `tmp, domain, expected` to be arrays.